incubator/hydra/templates/NOTES.txt

```diff
@@ -0,0 +1,16 @@
+{{- if eq .Values.config.system.secret "this_is_a_secret_change_me" }}
+######################################################################
+#                             WARNING                                #
+# You are using the default secret provided with the helm chart when #
+# issuing authorization tokens.  This is very insecure.              #
+######################################################################
+{{- end }}
+
+Hydra should not be facing the internet itself, so you can access hydra via. 
```

Comment:
Part of Hydra's authorization flow requires a redirect to Hydra. It definitely does need to be publicly accessible.

_Overview of the OAuth2 Authorization Code flow_
Basically, the user is initially redirected by the client to Hydra, which then redirects to the consent app with a signed challenge token. The consent app does necessary authentication and consent before redirecting the user back to Hydra (passing a signed challenge response). Hydra validates the response and generates the tokens before redirecting the user back to the client with the proper tokens.

This is incredibly simplified, but the key thing is that the user does need to be able to access Hydra from outside of the kubernetes cluster.